### PR TITLE
Refactor to use crawlee

### PR DESCRIPTION
**Pull request by Pijukatel**

### CODE REVIEW

**Comment by Pijukatel** [comment]
I tested actor here: https://console.apify.com/actors/hRZbnFhePGa88AhMQ/source

File: pyproject.toml [hunk 1]
```diff
@@ -9,12 +9,13 @@ package-mode = false
 
 [tool.poetry.dependencies]
 python = "^3.12"
```
[the remark this answers is not in the document]
Comment (Pijukatel) [reply pyproject.toml:11]: That would require to release new apify docker image for python3.13:
https://docs.apify.com/platform/actors/development/actor-definition/dockerfile#python-base-images

File: src/input_handling.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    Args:
+        page_function: The string representation of the user-defined function.
+
+    Returns:
+        The extracted user-defined function.
+
+    Raises:
+        KeyError: If the function name `USER_DEFINED_FUNCTION_NAME` cannot be found.
+    """
+    scope: dict = {}
```
[the remark this answers is not in the document]
Comment (Pijukatel) [reply src/input_handling.py:84]: I think that mypy has troubles with type inference from empty dict. So : dict was the minimal annotation that made mypy happy.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vdusek) [reply pyproject.toml:11]: let's skip it for now then, thanks

**Review by vdusek — APPROVED** [review]
LGTM

**Pull request merged**